### DISCUSSION TITLE
Python class inherit from PyO3 class

### DISCUSSION THREAD

**Discussion by jblindsay** [PyO3/pyo3#3624] [Questions] [answered]
Is it possible for a Python defined class to inherit from a PyO3 class? I've tried this and I receive an error indicating that, `builtins.WbEnvironment' is not an acceptable base type`. Is there perhaps a flag in my PyO3 project that I need to use to indicate that the class WbEnvironment is not final?

**Comment by adamreichold** [comment, marked as answer]
You need to add `#[pyclass(subclass)]`, c.f. <https://pyo3.rs/v0.20.0/class#customizing-the-class>

**Reply by jblindsay** [reply]
I figured it was something simple like that. Not sure how I missed that in the docs. Thanks for your help.
